src/bundle.c

```diff
@@ -906,14 +907,15 @@ static enum swupd_code install_bundles(struct list *bundles, struct list **subs,
 		string_or_die(&hashpath, "%s/staged/%s", state_dir, file->hash);
 
 		if (access(hashpath, F_OK) < 0) {
-			/* fallback to verify_fix_path below, which will check the hash
-			 * itself */
+			/* the file does not exist in the staged directory, it will need
+			 * to be downloaded again */
 			free_string(&hashpath);
 			continue;
 		}
 
+		/* make sure the file is not corrupt */
 		ret = verify_file(file, hashpath);
-		if (!ret) {
+		if (ret == false) {
```

Comment:
don't mix int and bool. It's very confusing. Use 

    if (!verify_file(...))

or create another bool variable for that.

Reply:
we were already mixing int and bool cause verify_file returns a bool. But I changed this to make it clearer so will change this back if it makes it more confusing.